### PR TITLE
add support for warehouse schemas

### PR DESCRIPTION
**Pull request by Tantalor93**

### CODE REVIEW

File: src/main/java/com/gooddata/warehouse/WarehouseSchema.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    public static final UriTemplate TEMPLATE = new UriTemplate(URI);
+
+    private String name;
+
+    private String description;
+
+    private Map<String, String> links;
+
+    @JsonCreator
+    public WarehouseSchema(@JsonProperty("name") String name,
```
Comment (martiner) [inline src/main/java/com/gooddata/warehouse/WarehouseSchema.java]: private?

File: src/test/java/com/gooddata/warehouse/WarehouseServiceIT.java [hunk 2]
```diff
@@ -30,11 +31,13 @@
 public class WarehouseServiceIT extends AbstractGoodDataIT {
 
     private static final String TITLE = "Test";
+    private static final String SCHEMA_NAME = "default";
```
Comment (martiner) [inline src/test/java/com/gooddata/warehouse/WarehouseServiceIT.java]: what is the default schema? is it something always present? maybe user would use some handy get method in the service?

File: src/main/java/com/gooddata/warehouse/WarehouseSchema.java (hunk 1 above)
Comment (Tantalor93) [inline src/main/java/com/gooddata/warehouse/WarehouseSchema.java]: changed to private

File: src/test/java/com/gooddata/warehouse/WarehouseServiceIT.java (hunk 2 above)
Comment (Tantalor93) [inline src/test/java/com/gooddata/warehouse/WarehouseServiceIT.java]: yes, default schema always exists in warehouse.  Added method `getDefaultWarehouseSchema` to the service.

File: src/main/java/com/gooddata/warehouse/WarehouseSchema.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class WarehouseSchema {
+
+    private static final String SELF_LINK = "self";
+    private static final String INSTANCE_LINK = "instance";
+
+    public static final String URI = WarehouseSchemas.URI + "/{name}";
+    public static final UriTemplate TEMPLATE = new UriTemplate(URI);
+
+    private String name;
```
Comment (martiner) [inline src/main/java/com/gooddata/warehouse/WarehouseSchema.java]: final?

**Pull request merged**